File: Algorithms/682. Baseball Game/javascript/solution.js
/**
 * @param {string[]} ops
 * @return {number}
 */
var calPoints = function (ops) {
    const stack = [];
    for (const op of ops) {
        switch (op) {
            case '+':
                stack.push(stack[stack.length - 1] + stack[stack.length - 2]);
                break;
            case 'C':
                stack.pop();
                break;
            case 'D':
                stack.push(stack[stack.length - 1] * 2);
                break;
            default:
                stack.push(parseInt(op));
                break;
        }
    }

    return stack.reduce((acc, cur) => acc + cur);
};

File: Algorithms/682. Baseball Game/python/solution.py
class Solution:
    def calPoints(self, ops: list[str]) -> int:
        stack = []
        for op in ops:
            match(op):
                case('+'):
                    stack.append(stack[-1] + stack[-2])
                case('C'):
                    stack.pop()
                case('D'):
                    stack.append(2 * stack[-1])
                case _:
                    stack.append(int(op))

        return sum(stack)
